rust/worker/src/segment/record_segment.rs

```diff
@@ -810,27 +813,24 @@ impl RecordSegmentReader<'_> {
         Ok(data)
     }
 
-    /// Returns all ids in the record segment sorted.
-    pub(crate) async fn get_all_user_ids(&self) -> Result<Vec<&str>, Box<dyn ChromaError>> {
-        let mut ids = Vec::new();
-        let max_size = self.user_id_to_id.count().await?;
-        for i in 0..max_size {
-            let res = self.user_id_to_id.get_at_index(i).await;
-            match res {
-                Ok((_, user_id, _)) => {
-                    ids.push(user_id);
-                }
+    /// Returns all offset_ids in the record segment sorted.
+    pub(crate) async fn get_all_offset_ids(&self) -> Result<Vec<u32>, Box<dyn ChromaError>> {
+        let offset_id_count = self.count().await?;
```

Comment:
nit: at this point we know exactly what capacity of Vec is required. I would prefer it was preallocated.

Reply:
Refactored this function with `for` loop for better readability. Also preallocated the result vector.